drivers/display/oled/oled.c

```diff
@@ -0,0 +1,320 @@
+	                   0x417f480444784000,
+	                   0x0000447d40000000,
+	                   0x000040443d000000,
+	                   0x417f101824424200,
+	                   0x0040417f40400000,
+	                   0x427e027c027e4000,
+	                   0x427e4402427c4000,
+	                   0x003c4242423c0000,
+	                   0x00417f4909090600,
+	                   0x00060909497f4100,
+	                   0x00427e4402020400,
+	                   0x00644a4a4a360000,
+	                   0x00043f4444200000,
+	                   0x00023e4040227e40,
+	                   0x020e3240320e0200,
+	                   0x021e6218621e0200,
+	                   0x4262140814624200,
+	                   0x0143453805030100,
+	                   0x004662524a466200,
+	                   0x0000083641000000,
+	                   0x0000007f00000000,
+	                   0x0000004136080000,
+	                   0x0018080810101800,
+	                   0xAA55AA55AA55AA55
+};
+
+/***************************************************************************//**
+ * @brief Turns display on/off.
+ *
+ * @param device - The device structure.
+ *        on_off - DISP_ON
+ *                 DISP_OFF
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t display_on_off(struct ssd_1306_dev *device, uint8_t on_off)
+{
+	int32_t ret = SUCCESS;
+	uint8_t command;
+
+	ret |= gpio_set_value(device->dc_pin, DC_CMD);
+
+	command = (on_off == DISP_ON) ? 0xAF : 0xAE;
+	ret |= spi_write_and_read(device->spi_desc, &command, 1U);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Moves cursor to desired position.
+ *
+ * @param device - The device structure.
+ *        row
+ *        column
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t move_cursor(struct ssd_1306_dev *device, uint8_t row, uint8_t column)
+{
+	int32_t ret = SUCCESS;
+	uint8_t command[3];
+	ret |= gpio_set_value(device->dc_pin, DC_CMD);
+
+	command[0] = 0x21;
+	command[1] = column*8;
+	command[2] = COL_NB * 8 - 1U;
+	ret |= spi_write_and_read(device->spi_desc, &command, 3U);
+
+	command[0] = 0x22;
+	command[1] = row;
+	command[2] = ROWS_NB - 1U;
+	ret |= spi_write_and_read(device->spi_desc, &command, 3U);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Clears data on display.
+ *
+ * @param device - The device structure.
+ *        row    - numbers of rows
+ *        column - numbers of columns
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t display_clear(struct ssd_1306_dev *device, uint8_t rows, uint8_t columns)
```

Comment:
Will this clear from position (0,0) or from current cursor position?
Is there any benefit here in using the 'display_print_ascii' API for more control?